gz-msgs-extras.cmake.in

```diff
@@ -0,0 +1,62 @@
+# Copyright 2023 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# copied from gz-msgs/gz-msgs-extras.cmake
+
+find_package(Python3 REQUIRED COMPONENTS Interpreter)
+
+include(${@PROJECT_NAME@_DIR}/gz_msgs_string_utils.cmake)
+include(${@PROJECT_NAME@_DIR}/gz_msgs_protoc.cmake)
+include(${@PROJECT_NAME@_DIR}/gz_msgs_factory.cmake)
+include(${@PROJECT_NAME@_DIR}/gz_msgs_generate.cmake)
+include(${@PROJECT_NAME@_DIR}/target_link_messages.cmake)
+
+set(@PROJECT_NAME@_INSTALL_PATH "${@PROJECT_NAME@_DIR}/@PROJECT_CMAKE_EXTRAS_PATH_TO_PREFIX@")
+cmake_path(NORMAL_PATH @PROJECT_NAME@_INSTALL_PATH OUTPUT_VARIABLE @PROJECT_NAME@_INSTALL_PATH)
+set(PROTOC_NAME "@PROJECT_NAME@_protoc_plugin")
+set(PROTO_SCRIPT_NAME "@PROJECT_NAME@_generate.py")
+set(FACTORY_SCRIPT_NAME "@PROJECT_NAME@_generate_factory.py")
+
+set(@PROJECT_NAME@_PROTO_PATH ${@PROJECT_NAME@_INSTALL_PATH}/share/protos)
+set(@PROJECT_NAME@_PROTO_GENERATOR_PLUGIN ${@PROJECT_NAME@_INSTALL_PATH}/bin/${PROTOC_NAME})
+set(@PROJECT_NAME@_PROTO_GENERATOR_SCRIPT ${@PROJECT_NAME@_INSTALL_PATH}/bin/${PROTO_SCRIPT_NAME})
+set(@PROJECT_NAME@_FACTORY_GENERATOR_SCRIPT ${@PROJECT_NAME@_INSTALL_PATH}/bin/${FACTORY_SCRIPT_NAME})
+
+function(gz_msgs_generate_messages)
```

Comment:
Can you document this function? I know there's good documentation in the READMEs in the example directory, but having at least the input parameters documented here and a pointer to the examples would be great.